e2e-tests/e2e-tests.ts

```diff
@@ -0,0 +1,59 @@
+const assert = require('assert')
+const puppeteer = require('puppeteer')
+
+let browser
+let page
+let host = process.env.E2E_TEST_ENVIRONMENT
+
+before(async () => {
+  browser = await puppeteer.launch()
+  page = await browser.newPage()
+})
+
+after(async () => {
+  await browser.close()
+})
+
+function delay (timeout) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, timeout)
+  })
+}
+
+describe('A non-matching journey', () => {
+  it('should succeed', async () => {
+    await journey().then(async function () {
+      const heading = await page.$eval('h2#heading-non-matching', e => e.innerHTML)
+      assert(heading.includes('Attributes for user with pid:'),'Actual: ' + heading)
+    })
+  }).timeout(10000)
+})
+
+describe('A matching journey', () => {
+  it('should succeed', async () => {
+    await journey(true).then(async function () {
+      // matching/user creation happening
+      await delay(2000)
```

Comment:
It's a shame there isn't a similar api to capybara for this where you can set a custom timeout on a page by page basis :(